src/__tests__/wait-for.js

```diff
@@ -34,6 +34,17 @@ test('uses generic error if there was no last error', async () => {
   expect(result).toMatchInlineSnapshot(`[Error: Timed out in waitFor.]`)
 })
 
+test('uses full stack error trace when showOriginalStackTrace present', async () => {
+  const error = new Error('Throws the full stack trace')
+  const result = await waitFor(
+    () => {
+      throw error
+    },
+    {timeout: 8, interval: 5, showOriginalStackTrace: true},
+  ).catch(e => e)
+  expect(result.stack).toBe(error.stack)
+})
+
```

Comment:
Could we add one test to verify that the stack trace is set correctly?

This might be tricky though. If you struggle with it, go ahead and push what you have and we can help :)

Reply:
This is especially interesting for different node versions in case node changes how they display stack traces.

Reply:
i am not knowing how to proceed .. any direction of help will be appreciated .. from what i understood you want me to add a new test that verifies the stack ?

Reply:
I think a test that shows the actual trace would be nice. You'd probably have to strip the parts of the filepaths that are tied to your local filesystem. You also want to only match the last significant stacks. Otherwise we're probably matching node internals.

Reply:
i am sorry i tried alot but i am not sure how to proceed .

Reply:
No worries @rahulchavan30! We can help with this part. Thanks for all you've done :)

Reply:
Thank you Kent and thank you so much for the opportunity I really got to learn so many things in short time of span

Reply:
That's wonderful to hear :) Take a look at my changes. It ended up being a bit challenging for me as well, but I'm happy with what we're going to ship together :)